x-pack/plugins/screenshotting/server/screenshots/observable.ts

```diff
@@ -181,13 +181,14 @@ export class ScreenshotObservableHandler {
 
     return defer(() => getNumberOfItems(driver, this.logger, waitTimeout, this.layout)).pipe(
       mergeMap((itemsCount) => {
-        // set the viewport to the dimentions from the job, to allow elements to flow into the expected layout
+        // set the viewport to the dimensions from the job, to allow elements to flow into the expected layout
         const viewport = this.layout.getViewport(itemsCount) || getDefaultViewPort();
 
-        return forkJoin([
-          driver.setViewport(viewport, this.logger),
-          waitForVisualizations(driver, this.logger, waitTimeout, itemsCount, this.layout),
-        ]);
+        return driver
+          .setViewport(viewport, this.logger)
+          .then(() =>
+            waitForVisualizations(driver, this.logger, waitTimeout, itemsCount, this.layout)
+          );
```

Comment:
Hi, I've just a side note, not related to this PR but something that can be considered in the future to improve the rendering time: what about considering setting the viewport before opening the page? this will skip at least 1 rendering cycle to resize all the visualizations. We can probably get this information from the URL and/or the saved object related to that dashboard/viz.

Reply:
Hey @markov00 ! That's a great idea. I was thinking we might have to do something a bit more complicated (like inject a resize observer) to really know when the reflow is done 😅 . Your suggestion sounds a lot simpler and definitely worth looking into.

Reply:
It seems the reason we do this here is because this is the point where we know how many items there are to render. For print layout, we have to know the number of items to resize the viewport: https://github.com/elastic/kibana/blob/main/x-pack/plugins/screenshotting/server/layouts/print_layout.ts#L47

Reply:
For a later PR perhaps: let's consider making this whole mergeMap conditional, and used only if the report job is the "optimized for print" layout mode. 

```
        if (this.layout.id === LayoutTypes.PRINT) {
          // set the viewport to the dimentions from the job, to allow elements to flow into print-optimized layout
          const viewport = this.layout.getViewport(itemsCount) || getDefaultViewPort();
          await driver.setViewport(viewport, this.logger);
        }
```

Reply:
That's a good point @tsullivan , it looks like the resize is for height in that case so I think we should at least be able to address a "width" reflow which seems could be more problematic.

I'll open up an issue to capture your suggested change!